barretenberg/cpp/src/barretenberg/benchmark/honk_bench/ultra_honk_rounds.bench.cpp

```diff
@@ -24,39 +24,41 @@ BBERG_PROFILE static void test_round_inner(State& state, honk::UltraProver& prov
     auto time_if_index = [&](size_t target_index, auto&& func) -> void {
         if (index == target_index) {
             state.ResumeTiming();
-            func();
+        }
+        func();
+        if (index == target_index) {
             state.PauseTiming();
-        } else {
-            func();
         }
     };
-    for (auto _ : state) {
-        state.PauseTiming();
-        time_if_index(PREAMBLE, [&] { prover.execute_preamble_round(); });
-        time_if_index(WIRE_COMMITMENTS, [&] { prover.execute_wire_commitments_round(); });
-        time_if_index(SORTED_LIST_ACCUMULATOR, [&] { prover.execute_sorted_list_accumulator_round(); });
-        time_if_index(GRAND_PRODUCT_COMPUTATION, [&] { prover.execute_grand_product_computation_round(); });
-        time_if_index(RELATION_CHECK, [&] { prover.execute_relation_check_rounds(); });
-        time_if_index(ZEROMORPH, [&] { prover.execute_zeromorph_rounds(); });
-        state.ResumeTiming();
-    }
+
+    time_if_index(PREAMBLE, [&] { prover.execute_preamble_round(); });
+    time_if_index(WIRE_COMMITMENTS, [&] { prover.execute_wire_commitments_round(); });
+    time_if_index(SORTED_LIST_ACCUMULATOR, [&] { prover.execute_sorted_list_accumulator_round(); });
+    time_if_index(GRAND_PRODUCT_COMPUTATION, [&] { prover.execute_grand_product_computation_round(); });
+    time_if_index(RELATION_CHECK, [&] { prover.execute_relation_check_rounds(); });
+    time_if_index(ZEROMORPH, [&] { prover.execute_zeromorph_rounds(); });
 }
 BBERG_PROFILE static void test_round(State& state, size_t index) noexcept
 {
     barretenberg::srs::init_crs_factory("../srs_db/ignition");
 
-    honk::UltraComposer composer;
-    // TODO(AD) benchmark both sparse and dense circuits?
-    honk::UltraProver prover =
-        bench_utils::get_prover(composer, &bench_utils::generate_keccak_test_circuit<UltraCircuitBuilder>, 1);
-    test_round_inner(state, prover, index);
+    for (auto _ : state) {
+        state.PauseTiming();
+        honk::UltraComposer composer;
+        // TODO(AD) benchmark both sparse and dense circuits?
```

Comment:
Could you make a GH issue in Barretenberg repo and reference here if you think this is something we should follow up with?